includes/class-sensei-course.php

```diff
@@ -1426,6 +1426,17 @@ public function filter_my_courses( $query ) {
 		_deprecated_function( __METHOD__, '3.0.0' );
 	}
 
+	/**
+	 * Has results links check.
+	 *
+	 * @param int $course_id Course ID.
+	 *
+	 * @return boolean
+	 */
+	private static function has_results_links( $course_id ) {
+		return ! empty( apply_filters( 'sensei_results_links', '', $course_id ) );
```

Comment:
I changed it from `has_filter` to `apply_filters` because it didn't work for the purpose it's used. Basically, it checks if something is added through that filter, and I think the filters is usually used with the check if they will be modified inside them.

We had an issue in the Certificates plugin, for example, where the filter was added, but in some cases, we didn't add anything and return the default value sent to the filter.

I believe it would be the safest way to do that. Do you have some other idea for that?

Reply:
I added this extra check: https://github.com/Automattic/sensei/pull/4322/commits/3b1bbdc94157e1ac85f85945099b92a7fb639298

WDYT?

Reply:
Adding a screenshot with the Certificates plugin to illustrate the problem that was fixed:

<img width="985" alt="Screen Shot 2021-10-05 at 11 14 07" src="https://user-images.githubusercontent.com/876340/136040813-6145175c-ffee-4794-9b75-2a80e9daa144.png">